tests/spin/test_spin_hamiltonian.py

```diff
@@ -782,3 +783,826 @@ def test_fermi_hubbard_hamiltonian_matrix(shape, n_cells, t, coulomb, expected_h
+            + -0.05 * (X(2) @ Z(3) @ Z(4) @ Z(5) @ X(6))
+            + -0.05 * (Y(3) @ Z(4) @ Z(5) @ Z(6) @ Y(7))
+            + -0.05 * (X(3) @ Z(4) @ Z(5) @ Z(6) @ X(7))
+            + -0.05 * (Y(4) @ Z(5) @ Y(6))
+            + -0.05 * (X(4) @ Z(5) @ X(6))
+            + -0.05 * (Y(5) @ Z(6) @ Y(7))
+            + -0.05 * (X(5) @ Z(6) @ X(7))
+            + 2.0 * I(0)
+            + -0.25 * Z(1)
+            + -0.25 * Z(0)
+            + -0.25 * (Z(0) @ Z(1))
+            + -0.5 * Z(3)
+            + -0.5 * Z(2)
+            + -0.75 * Z(5)
+            + -0.75 * Z(4)
+            + 0.25 * (Z(4) @ Z(5))
+            + -0.5 * Z(7)
+            + -0.5 * Z(6)
+            + 0.125 * (Z(0) @ Z(2))
+            + 0.125 * (Z(0) @ Z(3))
+            + 0.125 * (Z(1) @ Z(2))
+            + 0.125 * (Z(1) @ Z(3))
+            + 0.125 * (Z(0) @ Z(4))
+            + 0.125 * (Z(0) @ Z(5))
+            + 0.125 * (Z(1) @ Z(4))
+            + 0.125 * (Z(1) @ Z(5))
+            + 0.125 * (Z(2) @ Z(6))
+            + 0.125 * (Z(2) @ Z(7))
+            + 0.125 * (Z(3) @ Z(6))
+            + 0.125 * (Z(3) @ Z(7))
+            + 0.125 * (Z(4) @ Z(6))
+            + 0.125 * (Z(4) @ Z(7))
+            + 0.125 * (Z(5) @ Z(6))
+            + 0.125 * (Z(5) @ Z(7)),
+        ),
+    ],
+)
+def test_emery_hamiltonian_matrix(shape, n_cells, t, u, v, expected_ham):
+    r"""Test that the correct Emery Hamiltonian is generated when interaction parameters are provided as a matrix"""
+
+    emery_ham = emery(lattice=shape, n_cells=n_cells, hopping=t, coulomb=u, intersite_coupling=v)
+
+    qml.assert_equal(emery_ham, expected_ham)
+
+
+def test_hopping_error_haldane():
+    r"""Test that an error is raised when the shape of provided interaction parameters is wrong for
+    Haldane Hamiltonian."""
+    n_cells = [4, 4]
+    lattice = "Square"
+    with pytest.raises(
+        ValueError,
+        match=re.escape("The hopping1 parameter should be a constant or an array of shape (16,16)"),
+    ):
+        haldane(lattice=lattice, n_cells=n_cells, hopping1=[1.0, 2.0])
+
+    with pytest.raises(
+        ValueError,
+        match=re.escape("The hopping2 parameter should be a constant or an array of shape (16,16)"),
+    ):
+        haldane(lattice=lattice, n_cells=n_cells, hopping1=1.0, hopping2=[0.5, 0.6, 0.7])
+
+    with pytest.raises(
+        ValueError,
+        match=re.escape("The phi parameter should be a constant or an array of shape (16,16)"),
+    ):
+        haldane(lattice=lattice, n_cells=n_cells, hopping1=1.0, hopping2=0.1, phi=[0.5, 0.6])
+
+
+def test_mapping_error_haldane():
+    r"""Test that an error is raised when unsupported mapping is provided"""
+    n_cells = [4, 4]
+    lattice = "Square"
+    with pytest.raises(ValueError, match="The 'bk_sf' transformation is not available."):
+        haldane(lattice=lattice, n_cells=n_cells, mapping="bk_sf")
+
+
+@pytest.mark.parametrize(
+    # expected_ham here was obtained from datasets
```

Comment:
```suggestion
    # expected_ham here was obtained from manually
```
shouldn't this be manual?